src/cargo/util/command_prelude.rs

```diff
@@ -688,7 +688,14 @@ Run `{cmd}` to see possible targets."
         workspace: Option<&Workspace<'_>>,
         profile_checking: ProfileChecking,
     ) -> CargoResult<CompileOptions> {
-        let spec = self.packages_from_flags()?;
+        let w = self.flag("workspace");
+        let mut all = w || self.flag("all");
+        let package = self._values_of("package");
+        if w && !package.is_empty() {
+            all = false;
+        }
```

Comment:
This changes the behavior and is not even proposed in the original issue.

I would recommend starting with a warning first.

Reply:
here I am again, sorry for my lack of seriousness, i have a small question. issuing an error when the user uses `--package <package> & --workspace / --all` would be good for this issue? I have the impression that the fact that the package doesn't exist doesn't change anything.

Reply:
You can create a workspace and test it out. Before this patch, `cargo b -p foo --workspace` works without error. After this, it reports `foo` matching nothing. This should be a warning first, not a hard error.